app/models/booking.rb

```diff
@@ -2,4 +2,14 @@ class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :article
   has_one :review
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :start_must_be_before_end_date
+
+  private
+
+  def start_must_be_before_end_date
+    errors.add(:start_date, "must be before end time") unless
+        start_date < end_date
```

Comment:
The validation just looks at conditions and see whether they're checked. So here we compare start and end dates. I am adding a validation for start_date > Date.today